server/src/internalClusterTest/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java

```diff
@@ -1135,15 +1124,11 @@ private void checkMatchedFieldsCase(boolean requireFieldMatch, String type) thro
             .requireFieldMatch(requireFieldMatch);
         fooField.matchedFields("foo", "foo.plain");
         req = prepareSearch("test").highlighter(new HighlightBuilder().field(fooField));
-        assertResponse(
+        assertResponses(
+            response -> assertHighlight(response, 0, "foo", 0, equalTo("<em>running</em> with <em>scissors</em>")),
             req.setQuery(queryStringQuery("running scissors").field("foo")),
-            response -> assertHighlight(response, 0, "foo", 0, equalTo("<em>running</em> with <em>scissors</em>"))
-        );
-
-        // Now make half the matches come from the stored field and half from just a matched field.
-        assertResponse(
-            req.setQuery(queryStringQuery("foo.plain:running scissors").field("foo")),
-            response -> assertHighlight(response, 0, "foo", 0, equalTo("<em>running</em> with <em>scissors</em>"))
+            // Now make half the matches come from the stored field and half from just a matched field.
+            req.setQuery(queryStringQuery("foo.plain:running scissors").field("foo"))
```

Comment:
Wouldn't this change the existing `req` instance ? So we would end up running the 2nd test-case twice?

Reply:
You are right!! I missed this spot. I'll figure it out. Thank you Pano

Reply:
Fixed it